tools/c7n_gcp/c7n_gcp/filters/metrics.py

```diff
@@ -156,7 +156,8 @@ def batch_resources(self, resources):
         resource_filter = []
         batch_size = len(self.filter)
         for r in resources:
-            resource_name = self.manager.resource_type.get_metric_resource_name(r)
+            resource_name = self.manager.resource_type.get_metric_resource_name(
```

Comment:
this also needs to update extant implementations of this (sql, iam) to accept the extra parameter as well as the base class.